### PR TITLE
docs: add substrafl overview

### PR DESCRIPTION
**Pull request by RomainGoussault**
Add substrafl overview.

### CODE REVIEW

File: docs/source/substrafl_doc/index.rst [hunk 1]
```diff
@@ -1,4 +1,6 @@
 
+substrafl_overview.rst
```
Comment (Esadruhn) [inline docs/source/substrafl_doc/index.rst:2]: what about doing 

```suggestion
substrafl_overview
```

 or 

```
.. toctree::
   :glob:
   :caption: Substrafl overview
   :maxdepth: 2

   substrafl_overview
```

Comment (RomainGoussault) [reply docs/source/substrafl_doc/index.rst]: The second solution works, thanks!

File: docs/source/substrafl_doc/substrafl_overview.rst [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+There are three types of node:
+
+* TrainDataNode: one of the organizations the local training takes place on, with a set of data samples and an opener used for training.
+* TestDataNode: one of the organizations the model evaluation takes place on, with a set of data samples and an opener used for testing.
+* AggregationNode: the organization on which the aggregation, if there is one, takes place.
+
+
+Federated Learning Strategies
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+A FL strategy describes how to train a model on distributed data. The most well known strategy is the Federated Averaging strategy: train locally a model on every organization, then aggregate the weight updates from every organization, and then apply locally at each organization the averaged gradients. A strategy imposes some constraints on the model that can be used. For instance, you can use the Federated Averaging strategy with a deep neural network or with a logistic regression but not with a random forest. Several FL strategies are already implemented in Substrafl.
```
Comment (RomainGoussault) [inline docs/source/substrafl_doc/substrafl_overview.rst:73]: How can I add an internal link to the list of already implemented strategies https://connect-docs.owkin.com/en/latest/substrafl_doc/api/strategies.html ?

**Review by Fabien-GELUS — APPROVED** [review]
LGTM! (didnt test)

File: docs/source/substrafl_doc/substrafl_overview.rst (hunk 2 above)
Comment (RomainGoussault) [reply docs/source/substrafl_doc/substrafl_overview.rst]: it works thanks @Fabien-GELUS!

**Pull request merged**